crates/core_simd/src/first.rs

```diff
@@ -42,6 +47,44 @@ macro_rules! impl_vector {
                     self.0
                 }
             }
+
+            /// Loads a vector from a slice, without doing bounds checking
+            ///
+            /// # Safety
+            /// The length of `src` must be at least `LANES`.
+            #[inline]
+            pub unsafe fn load_from_slice_unchecked(src: &[$type]) -> Self {
+                (src.as_ptr() as *const Self).read_unaligned()
+            }
+
+            /// Loads a vector from a slice.
+            ///
+            /// # Panics
+            /// This function will panic if the length of `src` is not at least `LANES`.
+            #[inline]
+            pub fn load_from_slice(src: &[$type]) -> Self {
+                assert!(src.len() >= LANES, "slice must be at least as long as vector");
+                unsafe { Self::load_from_slice_unchecked(src) }
+            }
+
+            /// Stores a vector to a slice, without doing bounds checking
+            ///
+            /// # Safety
+            /// The length of `dst` must be at least `LANES`.
+            #[inline]
+            pub unsafe fn store_to_slice_unchecked(self, dst: &mut [$type]) {
+                (dst.as_mut_ptr() as *mut Self).write_unaligned(self)
+            }
+
+            /// Stores a vector to a slice.
+            ///
+            /// # Panics
+            /// This function will panic if the length of `dst` is not at least `LANES`.
+            #[inline]
+            pub fn store_to_slice(self, dst: &mut [$type]) {
```

Comment:
I am not sure these names are the way to go: we don't use "load" and "store", we use "read" and "write". This risks being _especially_ silly if it optimizes to something other than `STR`/`ST1`/etc.